Example/NYTViewController.m

```diff
@@ -23,101 +24,67 @@ typedef NS_ENUM(NSUInteger, NYTViewControllerPhotoIndex) {
 @interface NYTViewController () <NYTPhotosViewControllerDelegate>
 
 @property (weak, nonatomic) IBOutlet UIButton *imageButton;
-@property (nonatomic) NSArray *photos;
+@property (nonatomic) NYTPhotoViewerArrayDataSource *dataSource;
 
 @end
 
 @implementation NYTViewController
 
 - (IBAction)imageButtonTapped:(id)sender {
-    self.photos = [[self class] newTestPhotos];
-    
-    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithPhotos:self.photos initialPhoto:nil delegate:self];
+    self.dataSource = [self.class newTimesBuildingDataSource];
+
+    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithDataSource:self.dataSource initialPhoto:nil delegate:self];
 
     [self presentViewController:photosViewController animated:YES completion:nil];
     
-    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithPhotos:self.photos];
+    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
+//    [self switchDataSourceOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
```

Comment:
I'd like to demonstrate both the possibility of updating a single photo, and the possibility of completely changing out the data source.

The complete-data-source-change is surprising, though, if you're not expecting it. I wanted that to be optional.

I like the `BOOL` suggestion and will implement it shortly.